packages/mobile/src/transactions/TransactionFeed.tsx

```diff
@@ -38,137 +28,95 @@ interface StateProps {
   invitees: Invitees
   commentKey: string | null | undefined
   addressToE164Number: AddressToE164NumberType
-  standbyTransactions: StandbyTransaction[]
   recipientCache: NumberToRecipient
 }
 
+export type FeedItem = TransactionFeedFragment & {
+  status: TransactionStatus // for standby transactions
+}
+
 type Props = {
   kind: FeedType
   loading: boolean
   error: ApolloError | undefined
-  data: UserTransactionsData | undefined
-  standbyTransactions: StandbyTransaction[]
-  transactionFilter?: (tx: StandbyTransaction) => boolean
+  data: FeedItem[] | undefined
 } & StateProps
 
 const mapStateToProps = (state: RootState): StateProps => ({
   invitees: state.invite.invitees,
   commentKey: privateCommentKeySelector(state),
   addressToE164Number: state.identity.addressToE164Number,
-  standbyTransactions: state.transactions.standbyTransactions,
   recipientCache: recipientCacheSelector(state),
 })
 
-function exchangeFilter(tx: Transaction) {
-  return (
-    tx !== null &&
-    // Show exchange and gold transactions in exchange tab feed
-    (tx.type === TransactionTypes.EXCHANGE || resolveCurrency(tx.symbol) === CURRENCY_ENUM.GOLD)
-  )
-}
+export class TransactionFeed extends React.PureComponent<Props> {
+  static fragments = {
+    transaction: gql`
+      fragment TransactionFeed on Transaction {
+        ...ExchangeItem
+        ...TransferItem
+      }
 
-function defaultFilter(tx: Transaction) {
-  return (
-    tx !== null &&
-    // Show exchange and stableToken transactions in home feed
-    (tx.type === TransactionTypes.EXCHANGE || resolveCurrency(tx.symbol) !== CURRENCY_ENUM.GOLD)
-  )
-}
+      ${ExchangeFeedItem.fragments.exchange}
+      ${TransferFeedItem.fragments.transfer}
+    `,
+  }
 
-export class TransactionFeed extends React.PureComponent<Props> {
   renderItem = (commentKeyBuffer: Buffer | null) => ({
     item: tx,
   }: {
-    item: ExchangeTransaction | TransferTransaction
+    item: FeedItem
     index: number
   }) => {
-    const { kind, addressToE164Number, invitees, recipientCache } = this.props
-
-    if (tx.type === TransactionTypes.EXCHANGE) {
-      return (
-        <ExchangeFeedItem
-          status={tx.status ? tx.status : TransactionStatus.Complete}
-          showGoldAmount={kind === FeedType.EXCHANGE}
-          {...tx}
-        />
-      )
-    } else if (isTransferType(tx.type)) {
-      // Identify invite txs
-      if (tx.hasOwnProperty('comment')) {
-        if (tx.comment === SENTINEL_INVITE_COMMENT) {
-          if (tx.type === TransactionTypes.SENT) {
-            tx.type = TransactionTypes.INVITE_SENT
-          } else if (tx.type === TransactionTypes.RECEIVED) {
-            tx.type = TransactionTypes.INVITE_RECEIVED
-          }
+    const { addressToE164Number, invitees, recipientCache } = this.props
+
+    if (tx.hasOwnProperty('comment')) {
+      // @ts-ignore
```

Comment:
Is it possible to use`if (tx.__typename === 'TransactionTransfer') {` to avoid `// @ts-ignore` here?

Reply:
Yes, will change it 👍